### ISSUE TITLE
Hosted Agentic provider cannot authenticate on a fresh install: agent_builder_license_key is read in 4 places, written in none

### ISSUE THREAD

**Issue by renduples** [renduples/agent-builder-wp-plugin#136]
Found while running the first real prompt test against a deliberately clean install of lffci.org (zero prior Agent Builder data), as part of #133.

## The bug

`agent_builder_license_key` is **read in four places and written in none**:

```
includes/class-llm-client.php:1301        $body['user_id'] = get_option( 'agent_builder_license_key', '' );
includes/class-provider-registry.php:435  ! empty( $p['api_key'] ) && '' !== get_option( 'agent_builder_license_key', '' )
includes/class-provider-registry.php:421  (docblock describing the above)
library/tools/report_issue/tool.php:250   $license = get_option( 'agent_builder_license_key', '' );
```

```console
$ grep -rnE "(update_option|add_option|update_site_option)\s*\(\s*['\"]agent_builder_license_key" --include=*.php --include=*.js . | grep -v vendor
(no matches)
```

The "Connect to Agentic AI" flow in `includes/class-admin-ajax.php` receives the credential from the server, saves it into the providers table via `Provider_Registry::save_api_key( 'agentic', $api_key )` (`:1245`) and then sets eight other options (`:1254-1264`) — but never sets `agent_builder_license_key`.

## Why it matters

Two things depend on that option, and both fail on a fresh install:

1. **Requests are unauthenticated.** `LLM_Client` sends `user_id` from the option for the hosted `agentic` provider, so the relay receives an empty identity and rejects the call:

   > `This site is not activated for Agentic hosted AI (missing license key). Complete plugin setup to connect your account.`

2. **The admin funnels the user back to signup, forever.** `Provider_Registry::has_usable_provider()` treats `agentic` as unusable without it, and that gates `Admin_Menu_Handler::any_llm_configured()` (`:1693`), whose own docblock says it *"gates the full admin menu: when nothing is configured the user is funnelled to the Quick Start / signup page."*

So the loop is: complete signup → key is stored → option still empty → admin decides nothing is configured → user is sent back to Quick Start → repeat. The free hosted tier — the default path for a new user — cannot be brought online.

## Why we hadn't seen it

lffci.org had the option set from an older build, which masked it. It only surfaced after the box was reset to a genuine first-install state. With zero installs today, **every** install will be a fresh one, so this is on the critical path for submission.

## Reproduction

1. Fresh WordPress, activate Agent Builder, no prior plugin data.
2. Complete "Connect to Agentic AI".
3. `wp eval 'var_dump( get_option( "agent_builder_license_key" ) );'` → `false`.
4. Any chat request → the relay error above.

## Confirmed fix direction (not yet applied)

Setting the option to the same credential already stored on the provider row made the site work immediately — `has_usable_provider()` returned true and a real prompt test passed (15,989 tokens, real call, no cache hit):

```php
$p = Provider_Registry::get( 'agentic' );
update_option( 'agent_builder_license_key', $p['api_key'] );
```

That suggests the connect handler should persist the credential in **both** places, i.e. add an `update_option( 'agent_builder_license_key', $api_key )` beside the existing `save_api_key()` call. Worth confirming against the relay whether `user_id` is meant to be the same value as the API key or a distinct account identifier, since `class-admin-ajax.php:1233` already accepts either `api_key` or `license_key` from the server response — if the server can return both, they may be intended to be different.

Not fixing it blind: it is an auth flow and the server contract should decide the shape.

## Scope note

Unrelated to the prompt-testing work itself; that branch only surfaced it. The workaround above is currently applied to lffci.org so it has a working provider.

**Comment by renduples** [comment]
Fixed in `87570ff` on `feature/prompt-testing-harness`.

**Fixed at the read site, not the write site.** Four separate paths store an agentic key — signup (`class-admin-ajax.php:1245`), the setup wizard (`admin/setup.php:28`), the generic Settings → APIs save (`class-admin-ajax.php:994`) and emergency-stop clearing — so adding `update_option()` to the signup handler would have left the other three broken in exactly the same way.

New `Provider_Registry::get_license_key()` honours the `agent_builder_license_key` option when a site has it set, and otherwise falls back to the `agentic` provider row, which is where all four paths already write. The three readers (`LLM_Client`, `has_usable_provider()`, `report_issue`) now go through it, so there is one resolution rule instead of four copies of a `get_option()` call.

On the open question from the report — whether `user_id` is the API key or a distinct identifier — it is the same credential. `handle_agentic_signup()` accepts it from the server as either `api_key` or `license_key` (`:1233`) and stores it once. That was also confirmed empirically before writing the fix: setting the option to the provider row's key on the reset QA box made the relay accept requests and a real prompt test pass.

**Not written back into the option**, deliberately — that would put a second, unencrypted copy of a live credential in `wp_options` when the provider row is already encrypted at rest.

Five regression tests in `tests/unit/test-provider-license-key.php`, covering the resolver, the option override, the admin-menu gate that caused the signup loop, and an end-to-end assertion that `user_id` actually reaches the request body. They are real cover, not decoration: reverting the resolver to the old one-line option read fails three of the five.

```
OK (116 tests, 303 assertions)
```

phpcs unchanged — the remaining findings in the touched files are pre-existing and identical before and after.

No version bump (3.4.1 is unreleased, so it went into that changelog entry) and no deploy, per the standing instruction. lffci.org still has the manual workaround applied; it stays correct after this ships, since an explicitly-set option takes precedence and holds the same value.
